pkg/libdevfile/libdevfile.go

```diff
@@ -193,3 +199,83 @@ func GetEndpointsFromDevfile(devfileObj parser.DevfileObj, ignoreExposures []v1a
 	}
 	return endpoints, nil
 }
+
+// GetComponentResourceManifestContentWithVariablesResolved returns the full content of either a Kubernetes or an Openshift
+// Devfile component, either Inlined or referenced via a URI.
+// No matter how the component is defined, it returns
+// the content with all variables substituted using the global variables map defined in `devfileObj`.
+// An error is returned if the content references an invalid variable key not defined in the Devfile object.
+func GetComponentResourceManifestContentWithVariablesResolved(devfileObj parser.DevfileObj, devfileCmp interface{},
```

Comment:
Now we get the entire devfileObj as a parameter, you could pass the name of the component instead of its content

Reply:
Is there a simple way to retrieve a Component by its name in `devfileObj`? It is not clear in the docs, but are we sure component names are unique in the Devfile? (Maybe by both name and type?)

Or maybe I could avoid passing the `devfileObj`, and use the variables map as an argument instead. This is what I actually used `devfileObj` for, to substitute the variables using: `devfileObj.Data.GetDevfileWorkspaceSpec().Variables`

WDYT?

Reply:
Ah, I found a `Filter` field in `DevfileOptions` that could allow to retrieve Components by their name. But still not sure about the unicity.

Reply:
Just found our `ComponentsWithSameNameError` error type we already have in `libdevfile`. This seems to answer my previous question :)
So I will make the changes you suggested, using the component name instead.